addons/hr_expense/models/hr_expense.py

```diff
@@ -1078,11 +1078,20 @@ def action_sheet_move_create(self):
     def action_unpost(self):
         for sheet in self:
             move = sheet.account_move_id
+            payment = move._get_reconciled_payments()
             sheet.account_move_id = False
+
+            if (sheet.payment_mode == 'own_account' and payment):
+                payment.line_ids.remove_move_reconcile()
+                payment.unlink()
+
             if move.state == 'draft':
                 move.unlink()
             else:
-                move._reverse_moves(cancel=True)
+                move._reverse_moves(default_values_list=[{
+                    'name': '/',
+                    'ref': _("Reversal of: %s", move.ref),
```

Comment:
@william-andre, thanks for all the feedback :raised_hands: 

No more questions from my side, just an update. It was coming from reversed moves so I added these lines. Think it is fine now. 

Thanks again ^^